### PR TITLE
refactor: use did:key in packagers (for didcomm v1)

### LINKED ISSUE #1207: fix hardcode base58

### LINKED ISSUE #1604: Inlined keys for didcomm
Service entries of type `did-communication` in did documents must be `did:key`s as per [didcomm conventions](https://github.com/hyperledger/aries-rfcs/blob/master/features/0067-didcomm-diddoc-conventions/README.md#service-conventions)

### LINKED ISSUE #1659: Update PublicKey parsing routine to support did:key method
Currently, Aries Framework supports the following encodings for public key: publicKeyBase58, publicKeyHex, publicKeyPem, and publicKeyJwk. However public key can be represented with did:key method as well.
```json
{
    "@type": "somemessagetype",
    "~service": {
        "recipientKeys": ["8HH5gYEeNc3z7PYXmd54d4x6qAfCNrqQqEB3nS7Zfu7K"],
    }
}
```

In the example above, did:key method can be used in "recipientKeys" instead of the "naked" key:

```json
{
    "@type": "somemessagetype",
    "~service": {
        "recipientKeys": ["did:key:z6MkmjY8GnV5i9YTDtPETC2uUAW6ejw3nk5mXF5yci5ab7th"],
    }
}
```

For more details please check https://github.com/hyperledger/aries-rfcs/tree/master/features/0360-use-did-key.

### LINKED ISSUE #1847: Preload framework defaults with key vdri
As a framework user, I would like to have the vdri registry to be preloaded with did:key vdri, so the default Aries instance has did:key resolving capabilities.

### PR DESCRIPTION
**Pull request by Baha-sk**
This change removes Base58 key encoding for recipients and sender for did:key (execept for legacy packer which will remain using bsae58 encoding).
It also includes vdr peer did creation with all VM and KeyAgreement keys, not only the first key.
This change adds key resolvers for the packers, one for did:key resolution and another one used for the 1PU test vector verification purposes using a thirdparty store resolution.
Finally, this change also moves did:key fingerprint code referencing did.Doc into a separate subpackage (pkg/vdr/fingerprint/didfp) to avoid cyclic dependency with jwk fingerprint.

closes #1604
closes #1207
closes #1659
closes #1847

Signed-off-by: Baha Shaaban <baha.shaaban@securekey.com>

### CODE REVIEW

**Review by fqutishat — APPROVED** [review]

File: pkg/didcomm/protocol/mediator/service.go [hunk 1]
```diff
@@ -748,6 +750,9 @@ func (s *Service) setKeyUpdateResponseCh(msgID string, keyUpdateCh chan *Keylist
 }
 
 func (s *Service) ensureConnectionExists(connID string) error {
+	//nolint:forbidigo
+	fmt.Println(fmt.Sprintf("~~~~ get router connID in routeStore: "+routeConnIDDataKey, connID))
```
Comment (rolsonquadras) [inline pkg/didcomm/protocol/mediator/service.go:754]: Can we remove this fmt statement ? If needed, we can use logger instead.

**Review by rolsonquadras — APPROVED** [review]

File: pkg/didcomm/common/service/destination_default.go [hunk 2]
```diff
@@ -30,8 +31,12 @@ func CreateDestination(didDoc *diddoc.Doc) (*Destination, error) {
 		return nil, fmt.Errorf("create destination: no recipient keys on didcomm service block in diddoc: %+v", didDoc)
 	}
 
-	// TODO ensure recipient keys are did:key's
-	//  https://github.com/hyperledger/aries-framework-go/issues/1604
+	for i, k := range didCommService.RecipientKeys {
+		if !strings.HasPrefix(k, "did:") {
```
Comment (Moopli) [inline pkg/didcomm/common/service/destination_default.go:35]: This is testing against prefix `did:`, so I guess we're still accepting did-url key references, right?

**Review by Moopli — APPROVED** [review]
Noticed a few small places where we could refactor to clean up a bit, but those can come later.
LGTM :+1:

File: pkg/didcomm/common/service/destination_default.go (hunk 2 above)
Comment (Baha-sk) [reply pkg/didcomm/common/service/destination_default.go:35]: This is a check to ensure RecipientKeys are always didkeys

The didCommService is already updated to have RecipientKeys as did keys. This is only a validation as per the removed TODO comment above.

File: pkg/didcomm/protocol/mediator/service.go (hunk 1 above)
Comment (Baha-sk) [reply pkg/didcomm/protocol/mediator/service.go]: removed

File: pkg/didcomm/common/service/destination_default.go (hunk 2 above)
Comment (Baha-sk) [reply pkg/didcomm/common/service/destination_default.go:35]: > Noticed a few small places where we could refactor to clean up a bit, but those can come later.
> LGTM 👍

Indeed there's lot's of room for improvements.

**Comment by Baha-sk** [comment]
> Noticed a few small places where we could refactor to clean up a bit, but those can come later.
> LGTM 👍

Indeed there's lot's of room for improvements.

**Review by aholovko — APPROVED** [review]

**Pull request merged**